### DISCUSSION TITLE
Global locale doesn't change the entire language

### DISCUSSION THREAD

**Discussion by mister-teddy** [opensumi/core#1332] [Q&A] [answered]
I tried to set the global locale when the app initialized:

```ts
export class ProjectSwitcherContribution  implements ClientAppContribution {
  initialize() {
    setLanguageId('en-US');
  }
}
```

And it's working with the most of the IDE, excepts for these commands:

<img width="1112" alt="Screen Shot 2022-07-12 at 19 11 59" src="https://user-images.githubusercontent.com/29925961/178486970-bb26ce04-c115-4a5b-aab3-421f4e0593e4.png">

I'm not sure what is that module, and how can change the language of it to English? Sorry that I don't know Chinese, and my target users of the IDE neither!
Thank you!

**Comment by erha19** [comment]
@nguyenhongphat0  see the samples here https://github.com/opensumi/opensumi-module-samples/blob/main/example/src/browser/i18n/setup.ts, if you want to replace locale language, you should call the `registerLocalizationBundle` early.

**Reply by mister-teddy** [reply]
Sorry but I'm afraid it's not working! I used the `ide-electron` template and it also has the same setup (https://github.com/opensumi/ide-electron/blob/main/src/common/i18n/setup.ts), tried changing the call order of `registerLocalizationBundle` but it still didn't work for except that module above 😞

**Comment by bytemain** [comment]
this might be a bug, the i18n string is already in the codebase, but seem not work.

**Reply by mister-teddy** [reply]
Yes I also think so! Hopefully someone maintaining the module will notice this.

**Reply by bytemain** [reply]
> Yes I also think so! Hopefully someone maintaining the module will notice this.

Thanks for the feedback! I will look into it later~

**Comment by bytemain** [comment, marked as answer]
v2.19 should already fix this problem.

But in this version, we still have to set the language of Monaco separately:

https://github.com/opensumi/core/blob/main/packages/startup/entry/web/app.tsx#L5


In the top of your code entry, using the func: `setLocale`
